src/containers.js

```diff
@@ -38,19 +42,69 @@ const update = async (fileName, fileContents, install = false) => {
     return false;
   }
 
-  // write temp file, validate and then overwrite
   if (!await validComposeFile(fileContents)) {
     throw new Error(`Invalid docker-compose yml for file ${fileName}`);
   }
 
+  if (backupIsEnabled()) {
+    await backupComposeFiles();
+  }
+
   await overwriteComposeFile(filePath, fileContents);
   await dockerComposeCli.pull(filePath);
 
   return true;
 };
 
+
+/**
+ * Generates a new name containing date of creation and random characters to avoid most collisions
+ * @returns {string} Name for folder e.g. 2023-07-17_131438_817_2Tjx
+ */
+const genBackupFolderName = () => {
+  const filefriendlyDatetime = new Date()
+    .toISOString()
+    .replace('T', '_')
+    .replaceAll(':', '')
+    .replace('.', '_')
+    .replace('Z', ''); // e.g. 2023-07-17_131438_817
+
+  const randomText = crypto.randomBytes(3).toString('base64url'); // 4 ASCII characters per 3 byte
+
+  return `${filefriendlyDatetime}_${randomText}`;
+};
+
+/**
+ * Backup yml files in globally defined {dockerComposeFilePath} into {rootBackupFolder}
+ */
+const backupComposeFiles = async () => {
+  const backupFolder = path.resolve(rootBackupFolder, genBackupFolderName());
+  await fs.promises.mkdir(backupFolder, { recursive: true });
+
+  const files = await getFilesInFolder(dockerComposeFilePath);
+  for (const filename of files) {
+    if (filename.endsWith('.yml')) {
+      await fs.promises.copyFile(
+        path.resolve(dockerComposeFilePath, filename),
+        path.resolve(backupFolder, filename)
+      );
+    }
+  }
+};
+
+/**
+ * @param {string} dir Directory to check for files
+ * @returns {string[]} Files in directory
+ */
+const getFilesInFolder = async (dir) => {
+  const dirEntries = await fs.promises.readdir(dir, { withFileTypes: true });
+  return dirEntries
+    .filter((dirEntry) => dirEntry.isFile())
+    .map((dirEntry) => dirEntry.name);
```

Comment:
```suggestion
    .map(file => file.name);
```
At this point, we have only files, I think it's clearer like this 🤔